ssm/crypto.py

```diff
@@ -297,17 +298,25 @@ def verify_cert_path(certpath, capath, check_crls=True):
 
 
 def get_certificate_subject(certstring):
-    """Return the certificate subject's DN, in legacy openssl format."""
-    p1 = Popen(['openssl', 'x509', '-noout', '-subject'],
-               stdin=PIPE, stdout=PIPE, stderr=PIPE, universal_newlines=True)
-
-    subject, error = p1.communicate(certstring)
+    """Return the certificate subject's DN, in legacy openssl format.
 
-    if (error != ''):
+    In 3.4.0, this was updated to use PyOpenSSL to maintain compatibility with
+    Python 3.6 and later versions, along with OpenSSL 1.0.2 and 1.1.1.
+    """
+    try:
+        subject_x509name = OpenSSL.crypto.load_certificate(
+            type=OpenSSL.crypto.FILETYPE_PEM,
+            buffer=certstring
+            ).get_subject()
+    except Exception as error:
         log.error(error)
-        raise CryptoException('Failed to get subject: %s' % error)
+        log.error(CryptoException)
+        raise error
+
+    # Outputs DN in slash-separated format
+    subject = "".join("/{:s}={:s}".format(name.decode(), value.decode())
```

Comment:
Could you set this up as a function? As it might end up useful elsewhere and means we can unit test it.

Reply:
Most recent commit has moved this to a function.